compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -10563,69 +10565,37 @@ OMR::Z::TreeEvaluator::arraycmpHelper(TR::Node *node,
       // Threshold for when to generate a preemptive CLC to avoid pulling in 2 cache lines if the arrays differ towards the beginning
       uint32_t earlyCLCThreshold = 32;
 
-      isEndLabelNeeded = loopIters || needResultReg || length > (earlyCLCEnabled ? earlyCLCThreshold : TR_MAX_CLC_SIZE);
-
-      if (length <= (earlyCLCEnabled ? earlyCLCThreshold : TR_MAX_CLC_SIZE) && (isFoldedIf || !needResultReg))
+      if (isFoldedIf)
          {
-         // these are the only cases where no regDeps are needed as code will be just CLC/BRC with no internal control flow
-         regDeps = deps;
-         isCondLabelNeeded = false;
+         regDeps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(deps, 0, 2 /*for memory reference base registrers*/, cg);
```

Comment:
I notice that the [previous version](https://github.com/eclipse/omr/blob/df44968440fe35301e2bf55e77c8c82958a4df9b/compiler/z/codegen/OMRTreeEvaluator.cpp#L10590-L10596) of this part of the code would check whether `deps` is NULL.  Is there a reason why that's no longer needed?

```
         if (isFoldedIf)
            {
            if (deps == NULL)
               regDeps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, numDeps, cg);
            else
               regDeps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(deps, 0, numDeps, cg);
            }
```

Reply:
@hzongaro Thanks for mentioning this. I looked at the implementation and it seems that the constructor checks for null deps therefore it was redundant. Here is the constructor implementation: https://github.com/eclipse/omr/blob/410ef0d231421886b3300fefc92f38a159b6866c/compiler/z/codegen/OMRRegisterDependency.cpp#L166-L168
Please let me know if you think I should keep that explicit check.